src/main/webapp/WEB-INF/static/js/trialmanager/germplasm/germplasm.js

```diff
@@ -540,28 +540,36 @@
 					$rootScope.navigateToTab('germplasm', {reload: true});
 				};
 
-				$scope.showPopOverCheck = function(entryId, currentValue, studyEntryPropertyId) {
-					if(!studyStateService.hasGeneratedDesign()) {
+				$scope.showPopOverCheck = function(entryIds, currentValue, checkForGeneratedDesign) {
+					if(!checkForGeneratedDesign || !studyStateService.hasGeneratedDesign()) {
+						if(!Array.isArray(entryIds)) {
+							entryIds = [parseInt(entryIds)];
+						}
 						$uibModal.open({
 							templateUrl: '/Fieldbook/static/angular-templates/germplasm/changeStudyEntryTypeModal.html',
 							controller: "changeStudyEntryTypeCtrl",
 							size: 'md',
 							resolve: {
-								entryId: function () {
-									return entryId;
+								entryIds: function () {
+									return entryIds;
 								},
 								currentValue: function () {
 									return currentValue;
-								},
-								studyEntryPropertyId: function () {
-									return studyEntryPropertyId;
 								}
 							},
 							controllerAs: 'ctrl'
 						});
 					}
 				};
 
+				$scope.setEntryType = function() {
+					if ($scope.selectedItems.length === 0) {
+						showAlertMessage('', $.germplasmMessages.setEntryTypeSelectEntry);
+					} else {
+						$scope.showPopOverCheck($scope.selectedItems, null, false);
```

Comment:
I don't understand why we don't check for generated design when setting entry type by batch?